examples/palace-tonegen.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+# Example for tone generator
+# Copyright (C) 2020  Ngô Ngọc Đức Huy
+#
+# This file is part of palace.
+#
+# palace is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation, either version 3 of the License,
+# or (at your option) any later version.
+#
+# palace is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with palace.  If not, see <https://www.gnu.org/licenses/>.
+
+from argparse import Action, ArgumentParser
+from itertools import count, takewhile
+from enum import auto, Enum
+from math import sin
+from sys import stderr
+from time import sleep
+from typing import Iterable
+
+from palace import Buffer, Context, Device, Source
+
+CHUNK_LEN: int = 12000
+QUEUE_SIZE: int = 4
```

Comment:
due to GH-10, we will not be able to play from decoder directly, i.e. we don't need this